records_mover/check_db_conn_engine.py

```diff
@@ -0,0 +1,31 @@
+# flake8: noqa
+
+import sqlalchemy
+from typing import Union, Optional, Tuple
+import warnings
+
+
+def check_db_conn_engine(db: Optional[Union[sqlalchemy.engine.Engine,
+                                            sqlalchemy.engine.Connection]] = None,
+                         db_conn: Optional[sqlalchemy.engine.Connection] = None,
+                         db_engine: Optional[sqlalchemy.engine.Engine] = None) -> \
+    Tuple[Optional[Union[sqlalchemy.engine.Engine,
+                         sqlalchemy.engine.Connection]],
+          Optional[sqlalchemy.engine.Connection],
+          sqlalchemy.engine.Engine]:
+    if db:
+        warnings.warn("The db argument is deprecated and will be"
+                      "removed in future releases.\n"
+                      "Please use db_conn for Connection objects and db_engine for Engine"
+                      "objects.",
+                      DeprecationWarning)
+    if not (db or db_conn or db_engine):
+        raise ValueError("Either db, db_conn, or db_engine must be provided as arguments")
+    if isinstance(db, sqlalchemy.engine.Connection) and not db_conn:
+        db_conn = db
```

Comment:
does the connection have an `engine` attribute or getter? I'm wondering if it would be useful in the case where we're passed in a connection to also return its corresponding engine alongside the connection instead of None for the engine.

If it does have such an attribute i think we could add the guarantee that the returned engine is never none which could be handy for some callers maybe? deferring to you on this but an idea

Reply:
We always pass back an engine!